engine/classes/Elgg/Application.php

```diff
@@ -569,6 +572,62 @@ public static function upgrade() {
 	}
 
 	/**
+	 * Runs database migrations
+	 *
+	 * @throws InstallationException
+	 * @return bool
+	 */
+	public static function migrate() {
+		try {
+			$conf = self::elggDir()->getPath('engine/schema/settings.php');
+			if (!$conf) {
+				throw new Exception('Settings file is required to run database migrations.');
+			}
+
+			$app = new \Phinx\Console\PhinxApplication();
+			$wrapper = new \Phinx\Wrapper\TextWrapper($app, [
+				'configuration' => $conf,
+			]);
+			$log = $wrapper->getMigrate();
+			error_log($log);
+		} catch (Exception $e) {
+			error_log($e->getMessage());
+			error_log($e->getTraceAsString());
+			throw new InstallationException($e->getMessage(), $e->getCode());
+		}
+
+		return true;
+	}
+
+	/**
+	 * Returns configuration array for database migrations
+	 * @return array
+	 */
+	public static function getMigrationSettings() {
+		$config = self::create()->services->config;
+		$config->loadSettingsFile();
```

Comment:
Note to self: I'll need to rewrite this function a bit for #11092. Something like:

```php
$config = self::factory()->services->config;
...
$config->dbprefix; // getVolatile going away
```